RHDEVDOCS-3614 ODC Release Notes, New features for 4.10 (#42818)
openshift/openshift-docs: release_notes/ocp-4-10-release-notes.adoc

[id="ocp-4-10-web-console"]
=== Web console

[id="ocp-4-10-web-console-developer-perspective"]
==== Developer perspective
* With this update, you can specify the name of a service binding connector in the *Topology* view while making a binding connection.
* With this update, you can see when the pipeline has been created but not started by looking at the status of the pod in the *Topology* view.
* This enhancement opens a dialog or alert that displays the export logs while an application export is in progress. You can use the dialog to cancel or restart the exporting process.
* With this update, you can add your new Helm Chart Repository to the Developer Catalog by creating a custom resource. Refer to the quick start guides in the Developer perspective to add a new *ProjectHelmChartRepository*.

> * With this update, you can view and use devfiles samples from (link:http://registry.devfile.io/[http://registry.devfile.io/]). These samples are present in the Developer Catalog.

Author: Done

[id="ocp-4-10-web-console"]
=== Web console

[id="ocp-4-10-web-console-developer-perspective"]
==== Developer perspective

> * You can now specify a name to a Service Binding connector in the *Topology* view while making a binding connection.

Reviewer: ```suggestion
* With this update, you can specify the name of a service binding connector in the *Topology* view while making a binding connection.
```
Lower case "service binding" unless you're talking about the "Service Binding Operator" proper noun.

Author: Thanks for pointing out the lower case. Will add this to the updates.

> * You can check if a pipeline is created and not started using the status decorator of the pod in the *Topology* view.

Author: @rolfedh  Your rephrase seems fine in the suggestion. "The pod" is enough also.

> * This enhancement gives you a better user experience when an application export is in progress a dialog box window or an alert will appear to view the logs, you can also cancel or restart the exporting process from this dialog box window.

Reviewer: ```suggestion
* This enhancement opens a dialog or alert that displays the export logs while an application export is in progress. You can use the dialog to cancel or restart the exporting process.
```

Author: Good suggestion.

> * You can now add your new Helm Chart Repository in the Developer Catalog by creating a new custom resource. Refer to the QuickStart guides in the Developer perspective to add a new *HelmChartRepo*.

Reviewer 2: Is this about namespaced helm chart repos? Then I would suggest to call the exact resource at the end:

```suggestion
* You can now add your new Helm Chart Repository in the Developer Catalog by creating a new custom resource. Refer to the QuickStart guides in the Developer perspective to add a new *ProjectHelmChartRepository*.
```

Author: Yes @jerolimov it is about namespaced helm chart repos. I'll update the text with the exact resource.

> * You can now add your new Helm Chart Repository in the Developer Catalog by creating a new custom resource. Refer to the QuickStart guides in the Developer perspective to add a new *HelmChartRepo*.

Reviewer: ```suggestion
* With this update, you can add your new Helm Chart Repository to the Developer Catalog by creating a custom resource. Refer to the QuickStart guides in the Developer perspective to add a new *HelmChartRepo*.
```

Author: Done

> * You can now view and use devfiles samples from (link:http://registry.devfile.io/[http://registry.devfile.io/]). These samples are present in the Developer Catalog.

Reviewer: ```suggestion
* With this update, you can view and use devfiles samples from (link:http://registry.devfile.io/[http://registry.devfile.io/]). These samples are present in the Developer Catalog.
```

Author: Done

[id="ocp-4-10-web-console"]
=== Web console

[id="ocp-4-10-web-console-developer-perspective"]
==== Developer perspective
* With this update, you can specify the name of a service binding connector in the *Topology* view while making a binding connection.
* With this update, creating pipelines workflow has now been enhanced:
** You can see when the pipeline has been created but not started by looking at the status of the pod in the *Topology* view.

> ** You can create an application using the *Import from Git* workflow, along with a template-generated pipeline by selecting the *Add pipeline* option. The webhook resources and URL are visible in the side panel of the selected resources.

Reviewer 3: ```suggestion
**You can now choose an user defined pipeline from a dropdown list while importing your application from import from Git/Docker pipeline workflow.
**Default webhooks are added for the pipelines that are created using Add flow and URL is visible in the side panel of the selected resources.
```

Author: Thanks for the suggestion @karthikjeeyar

> ** You can opt out of the default Tekton Hub integration by setting the parameter `enable-devconsole-integration` to `false` in the `TektonConfig` custom resource. When you opt out, only the platform compatible cluster tasks for your namespace are listed in the *Pipeline builder* -> *Add tasks* view.

Reviewer 4: @karthikjeeyar can you please verify if this is technically correct?

> ** You can opt out of the default Tekton Hub integration by setting the parameter `enable-devconsole-integration` to `false` in the `TektonConfig` custom resource. When you opt out, only the platform compatible cluster tasks for your namespace are listed in the *Pipeline builder* -> *Add tasks* view.

Reviewer 3: I Suggest the following changes, Please update the text as you see fit.

```suggestion
** Pipeline builder will contain the Tekton Hub tasks that are supported by the cluster's Architecture, all other unsupported tasks are excluded from the list.
** You can opt out of the default Tekton Hub integration by setting the parameter `enable-devconsole-integration` to `false` in the `TektonConfig` custom resource.
```

Author: Thanks for the suggestion @karthikjeeyar I'll incorporate these changes in the next edit.
